src/itkParseTileConfiguration.cxx

```diff
@@ -0,0 +1,204 @@
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkParseTileConfiguration.h"
+
+#include <cassert>
+#include <fstream>
+#include <limits>
+#include <sstream>
+
+#include "double-conversion.h"
+
+namespace // annonymous namespace
+{
+std::string
+getNextNonCommentLine( std::istream& in )
+{
+  std::string temp;
+  while ( std::getline( in, temp ) )
+    {
+    // this is neither an empty line nor a comment
+    if ( !temp.empty() && temp[0] != '#' )
+      {
+      break;
+      }
+    }
+  return temp;
+}
+
+static double_conversion::StringToDoubleConverter stringConverter(
+  double_conversion::StringToDoubleConverter::ALLOW_TRAILING_JUNK |
+  double_conversion::StringToDoubleConverter::ALLOW_LEADING_SPACES |
+  double_conversion::StringToDoubleConverter::ALLOW_TRAILING_SPACES,
+  0.0, std::numeric_limits<double>::quiet_NaN(), nullptr, nullptr);
+
+template< unsigned Dimension >
+itk::Tile< Dimension >
+parseLine( const std::string line, std::string& timePointID )
+{
+  itk::Tile< Dimension > tile;
+  std::stringstream ss( line );
+  std::string temp;
+
+  std::getline( ss, temp, ';' );
+  tile.FileName = temp;
+  std::getline( ss, temp, ';' );
+  if (timePointID.empty())
+    {
+    timePointID = temp;
+    }
+  else
+    {
+    itkAssertOrThrowMacro( temp == timePointID,
+      "Only a single time point is supported. " << timePointID << " != " << temp );
+    }
+  std::getline( ss, temp, '(' );
+
+  using PointType = itk::Point< double, Dimension >;
+  PointType p;
+  for (unsigned d = 0; d < Dimension; d++)
+    {
+    std::getline( ss, temp, ',' );
+    int processed = 0;
+    p[d] = stringConverter.StringToDouble( temp.c_str(), temp.length(), &processed );
+    }
+  tile.Position = p;
+
+  return tile;
+}
+
+template< unsigned Dimension >
+std::vector< itk::Tile< Dimension > >
+parseRow( std::string& line, std::istream& in, std::string& timePointID )
+{
+  std::vector< itk::Tile< Dimension > > row;
+
+  std::string timePoint;
+  itk::Tile< Dimension > tile = parseLine< Dimension >( line, timePoint );
+  row.push_back( tile );
+  line = getNextNonCommentLine( in );
+
+  while (in)
+    {
+    tile = parseLine< Dimension >( line, timePointID );
+    if ( tile.Position[0] < row.back().Position[0] ) // this is start of a new row
+      {
+      return row;
+      }
+    row.push_back( tile );
+    line = getNextNonCommentLine( in );
+    }
+
+  return row;
+}
+
+static double_conversion::DoubleToStringConverter doubleConverter(
+  double_conversion::DoubleToStringConverter::NO_FLAGS,
+  nullptr, nullptr, 'e', 0, 17, 1, 0 );
+
+} // annonymous namespace
+
+
+namespace itk
+{
+std::vector< std::vector< Tile< 2 > > >
+ParseTileConfiguration2D( const std::string pathToFile )
+{
+  constexpr unsigned Dimension = 2;
+  using PointType = itk::Point< double, Dimension >;
+  using TileRow = std::vector< Tile< Dimension > >;
```

Comment:
Please use these named types in the class and in the example so their meaning is known. `std::vector< std::vector< itk::Tile< 2 > >` was opaque.